src/frontend/e2e/utils/sample.ts

```diff
@@ -8,109 +14,113 @@ const defaultSequence =
+        getADateInThePast(minCollectionDays, maxCollectionDays, collectionDate)
+      ) as string,
+    },
+    sample: {
+      collection_date: collectionDate as string,
+      location_id: getValueOrDefault(
+        defaults?.sample?.location_id,
+        locationId
       ) as number,
-      lineage: {
-        confidence: "",
-        last_updated: GeneralUtil.getADateInThePast(),
-        lineage: sample(lineages) as string,
-        qc_status: "pass",
-        scorpio_call: "Omicron (BA.1-like)",
-        scorpio_support: 0.93,
-        version: "PUSHER-v1.13",
-      },
-      private: GeneralUtil.getValueOrDefault(
-        defaults?.private,
-        true
+      private: getValueOrDefault(
+        defaults?.sample?.private,
+        sample(trueOrFalse)
       ) as boolean,
-      private_identifier: GeneralUtil.generatePrivateSampleId(),
-      public_identifier: GeneralUtil.generatePublicSampleId(),
-      sequencing_date: GeneralUtil.getADateInThePast(),
-      submitting_group: {
-        id: 74,
-        name: "QA Automation",
-      },
-      upload_date: GeneralUtil.getADateInThePast(),
-      uploaded_by: {
-        id: 108,
-        name: "Playwright",
-      },
-    };
-  }
+      private_identifier: getValueOrDefault(
+        defaults?.sample?.private_identifier,
+        generatePrivateSampleId()
+      ) as string,
+      public_identifier: getValueOrDefault(
+        defaults?.sample?.public_identifier,
+        generatePublicSampleId()
+      ) as string,
+    },
+  };
+}
+
+/**
+ * method creates synthetic data for stubbing get Sample api response
+ * @param defaults - user supplied sample data to be included
+ * @param maxDays  - specifies earliest day of the sample, defaults 10
+ * @param minDays  - specifies lates day of the sample default 0 (today).
+ * @returns GetSampleResponseData
+ */
+export function getSampleResponseData(
+  defaults?: Partial<SampleResponseDefaults>,
+  minDays = 0,
+  maxDays = 10
+): SampleResponseData {
+  return {
+    collection_date: getADateInThePast(minDays, maxDays),
+    collection_location: {
+      country: "USA",
+      division: "California",
+      id: getValueOrDefault(
+        defaults?.collection_location,
+        locationId
+      ) as number,
+      location: "Corodano",
+      region: "California",
+    },
+    czb_failed_genome_recovery: getValueOrDefault(
+      defaults?.czb_failed_genome_recovery,
+      sample(trueOrFalse)
+    ) as boolean,
+    gisaid: {
+      gisaid_id: "",
+      status: "Not Found",
+    },
+    id: getValueOrDefault(defaults?.id, getRandomNumber()) as number,
+    lineage: {
+      confidence: "",
+      last_updated: getADateInThePast(),
+      lineage: getValueOrDefault(defaults?.lineage, sample(lineages)) as string,
+      qc_status: "pass",
+      scorpio_call: "Omicron (BA.1-like)",
+      scorpio_support: 0.93,
+      version: "PUSHER-v1.13",
+    },
+    private: getValueOrDefault(defaults?.private, true) as boolean,
```

Comment:
again, type assertion should not be here.

Reply:
@mmmmmmaya pls advise how I can silence TS/VSCODE when they error/warn on this. I can change type to any but would prefer better solution.

Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.ts(2322)

Reply:
Fixed